src/tools/yarprobotstatepublisher/src/robotstatepublisher.cpp

```diff
@@ -229,14 +229,7 @@ void YARPRobotStatePublisherModule::onRead(yarp::rosmsg::sensor_msgs::JointState
     iDynTree::JointIndex jntIndex;
     for (size_t i=0; i < v.name.size(); i++)
     {
-        for (size_t j=0; j < model.getNrOfDOFs(); j++)
-        {
-            if (v.name[i] == model.getJointName(j))
-            {
-                jntIndex = model.getJointIndex(v.name[i]);
-                break;
-            }
-        }
+        jntIndex = model.getJointIndex(v.name[i]);
 
```

Comment:
You should check that the joint does not have 0 dof, otherwise the line
~~~
 m_jointPos(model.getJoint(jntIndex)->getDOFsOffset()) = v.position[i];
~~~
will be an invalid memory access.

Reply:
Thanks, fixed with https://github.com/robotology/idyntree/pull/538/commits/328233115dd4e937ec05bb5b6848cfdd6948c919.